src/librustc_mir/build/mod.rs

```diff
@@ -238,6 +238,76 @@ pub fn construct<'a,'tcx>(hir: Cx<'a,'tcx>,
     )
 }
 
+
+///////////////////////////////////////////////////////////////////////////
+/// the main entry point for building MIR for an arbitrary expression
+
+pub fn construct_expr<'a,'tcx>(hir: Cx<'a,'tcx>,
+                          id: ast::NodeId,
+                          span: Span,
+                          expr: &'tcx hir::Expr)
+                          -> (Mir<'tcx>, ScopeAuxiliaryVec) {
+    let tcx = hir.tcx();
+    let cfg = CFG { basic_blocks: vec![] };
+    let ty = tcx.node_id_to_type(id);
+
+    let mut builder = Builder {
+        hir: hir,
+        cfg: cfg,
+        fn_span: span,
+        scopes: vec![],
+        scope_datas: vec![],
+        scope_auxiliary: ScopeAuxiliaryVec { vec: vec![] },
+        loop_scopes: vec![],
+        temp_decls: vec![],
+        var_decls: vec![],
+        var_indices: FnvHashMap(),
+        unit_temp: None,
+        cached_resume_block: None,
+    };
+
+    assert_eq!(builder.cfg.start_new_block(), START_BLOCK);
+    assert_eq!(builder.cfg.start_new_block(), END_BLOCK);
+
+    let call_site_extent = tcx.region_maps.item_extent(id);
```

Comment:
For constants and statics, item. For arbitrary expressions the extent of the binding to which the expression value is bound to.